plugins/pencilblue/controllers/actions/admin/sites/delete_site.js

```diff
@@ -53,8 +53,8 @@ module.exports = function DeleteSiteActionModule(pb) {
         var self = this;
         var siteid = self.pathVars.siteid;
         this.service.deleteSingle({where: {uid: siteid}}, function (err, site) {
-            if (util.isError(err)) {
-                return cb(err);
+            if (util.isError(err)|| !site) {
+                err = new Error("Failed to retrieve site from database for deletion");
```

Comment:
We would prefer a 404 when the requested site to delete does not exist.  This aligns better with the direction we are taking the newer APIs.